clash-lib/prims/vhdl/GHC_Num_Integer.primitives

```diff
@@ -189,16 +189,28 @@ end block;
   }
 , { "BlackBox" :
     { "name"      : "GHC.Num.Integer.integerShiftR#"
-    , "kind"      : "Expression"
-    , "type"      : "integerShiftR :: Integer -> Word# -> Integer"
-    , "template"  : "shift_right(~ARG[0],to_integer(~ARG[1]))"
+    , "kind"      : "Declaration"
+    , "type"      : "integerShiftR# :: Integer -> Word# -> Integer"
+    , "template"  :
+"-- INTEGER_SHIFT_R
+~RESULT <= shift_right(~ARG[0], to_integer(~VAR[count][1](30 downto 0)))
+    -- pragma translate_off
+    when (~ARG[1] >= to_unsigned(0, ~SIZE[~TYP[1]]-1)) else (others => 'X')
```

Comment:
```suggestion
    when (~ARG[1] >= 0) else (others => 'X')
```
When we were debugging this, I didn't really know what was going on. But given that just comparing against the `integer` 0 seems to work for the other primitives, let's simplify the ones that shift by a `Word#` amount.